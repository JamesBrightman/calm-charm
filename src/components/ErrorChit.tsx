import { FC, HTMLAttributes } from "react";
import { MdError } from "react-icons/md";

export interface ErrorChitProps extends HTMLAttributes<HTMLDivElement> {}

export const ErrorChit: FC<ErrorChitProps> = ({
  children,
  className,
  ...props
}) => {
  return (
    <div
      className={
        "flex flex-row gap-2 items-center justify-center text-center bg-red-200 p-2 rounded-lg text-black font-semibold" +
        " " +
        className
      }
      {...props}
    >
      <MdError className="w-8 h-8 text-red-500" />
      {children}
    </div>
  );
};
