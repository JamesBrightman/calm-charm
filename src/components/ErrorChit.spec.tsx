import React from 'react';
import { render, screen } from '@testing-library/react';
import { ErrorChit } from './ErrorChit';

describe('Error Chit Component', () => {
  it('Error Chit renders', () => {
    const { baseElement } = render(
      <ErrorChit>
        <p>{'Foo Bar'}</p>
      </ErrorChit>
    );

    expect(baseElement).toBeTruthy();
  });

  it('Error Chit matches snapshot', () => {
    const { baseElement } = render(
      <ErrorChit>
        <p>{'Foo Bar'}</p>
      </ErrorChit>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Error Chit displays custom text', () => {
    render(
      <ErrorChit>
        <p>{'Foo Bar'}</p>
      </ErrorChit>
    );

    expect(screen.getByText('Foo Bar')).toBeInTheDocument();
  });

  it('Error Chit renders with custom class', () => {
    render(
      <ErrorChit className="bg-blue-500 text-lg" data-testId="chitTest">
        <p>{'Foo Bar'}</p>
      </ErrorChit>
    );

    expect(screen.getByTestId('chitTest')).toHaveClass('bg-blue-500 text-lg');
  });
});
