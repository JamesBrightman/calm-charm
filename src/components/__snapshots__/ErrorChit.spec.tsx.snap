// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Chit Component Error Chit matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-row gap-2 items-center justify-center text-center bg-red-200 p-2 rounded-lg text-black font-semibold undefined"
    >
      <svg
        class="w-8 h-8 text-red-500"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <p>
        Foo Bar
      </p>
    </div>
  </div>
</body>
`;
